### PR TITLE
(DOCSP-8761): Port auth provider: Email/Password

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
[JIRA](https://jira.mongodb.org/browse/DOCSP-8761)

[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/realm/pegasus/DOCSP-8761-4-realsies/authentication/email-password.html)

### CODE REVIEW

File: source/authentication/email-password.txt [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+.. _ios-universal-links: https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html
+
+.. _email-password-authentication-overview:
+
+Overview
+--------
+
+The Email/Password authentication provider allows users to register
+accounts and log in using their email address and a password. Realm
+must confirm Email/Password users before it allows them to login to
```
Comment (cbush) [inline source/authentication/email-password.txt]:
```suggestion
must confirm Email/Password users before it allows them to log in to
```

File: source/authentication/email-password.txt [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+- :guilabel:`Email Confirmation URL`: the base of the URL included in
+  every confirmation email. Realm appends a unique ``token`` and
+  ``tokenId`` to this URL as query parameters to create a unique link
+  for every confirmation. To confirm the user, extract these query
+  parameters from the user's unique URL and pass the token and tokenId
+  to the Client SDK's ``confirmUser`` function.
+
+- :guilabel:`Email Confirmation Subject`: the subject of the email sent
+  to the user containing the confirmation URL. This value is optional:
+  if omitted, Stich uses a default subject line instead. Custom email
```
Comment (cbush) [inline source/authentication/email-password.txt]:
```suggestion
  if omitted, Realm uses a default subject line instead. Custom email
```

File: source/authentication/email-password.txt [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+- :guilabel:`Password Reset URL`: the base of the URL included in every
+  password reset email. Realm appends a unique ``token`` and
+  ``tokenId`` to this URL as query parameters to create a unique link
+  for every password reset. To reset the user's password, extract these
+  query parameters from the user's unique URL and pass the token and
+  tokenId to the Client SDK's ``resetPassword`` function.
+
+- :guilabel:`Reset Password Email Subject`: the subject line of the
+  email that Realm sends to users when they request to reset their
+  password. This value is optional: if omitted, Stich uses a default
```
Comment (cbush) [inline source/authentication/email-password.txt]:
```suggestion
  password. This value is optional: if omitted, Realm uses a default
```

File: source/authentication/email-password.txt [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+.. note::
+
+   If you configure the provider to use a password reset function then
+   the client SDK ``sendResetPasswordEmail()`` function will return an
+   error. You may still need to call ``resetPassword()`` if the password
+   reset function returns a ``pending`` status.
+
+The custom password reset function signature is
+:wikipedia:`variadic <Variadic_function_>`: it accepts any number of
+parameters. The first of these parameters is always an object containing
```
Comment (cbush) [inline source/authentication/email-password.txt]: This sentence ("The custom password ...") is overwritten... I would just remove this.

Comment (nathan-contino-mongo) [reply source/authentication/email-password.txt]: I'm not sure there's any other way for a user to understand that they can pass extra parameters to this function, is there?

File: source/authentication/email-password.txt (hunk 3 above)
Comment (nathan-contino-mongo) [reply source/authentication/email-password.txt]: Misspelled Stitch thrwarted my find&replace. Shucks.

**Pull request merged**